### PR TITLE
[SYCL] Make default device the one selected by default_selector

### PR DESCRIPTION
**Pull request by s-kanaev**
Host device still can be retrieved with help of special tag `host_device`:

    sycl::device D(sycl::host_device{});

### CODE REVIEW

**Comment by s-kanaev** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/845

File: sycl/include/CL/sycl/device.hpp [hunk 1]
```diff
@@ -28,13 +28,19 @@ class device_impl;
...
+struct host_device {};
+
 /// The SYCL device class encapsulates a single SYCL device on which kernels
 /// may be executed.
 ///
 /// \ingroup sycl_api
 class __SYCL_EXPORT device {
 public:
   /// Constructs a SYCL device instance as a host device.
+  device(host_device);
```
Comment (bader) [inline sycl/include/CL/sycl/device.hpp:41]: I don't see this constructor in the specification. Why do you think we should add it?

**Review by bader — CHANGES_REQUESTED** [review]
Please, make similar update to the platform class as well.

File: sycl/include/CL/sycl/device.hpp (hunk 1 above)
Comment (s-kanaev) [reply sycl/include/CL/sycl/device.hpp:41]: I don't think there's an approval to remove host device completely. Also, libsycl.so makes use of host device.

**Comment by s-kanaev** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/845

File: sycl/include/CL/sycl/device.hpp (hunk 1 above)
Comment (bader) [reply sycl/include/CL/sycl/device.hpp:41]: I'm not suggesting we move the host device support completely, but we should not add silently add non-standard functionality w/o extending the specification.

You can use `aspect_selector` with `host` aspect to select a host device. The host aspect is not in the standard, but it's already supported by DPC++, so no new extensions are needed.

**Pull request closed without merging**